File: Daily/944.delete-columns-to-make-sorted.js
/*
 * @lc app=leetcode id=944 lang=javascript
 *
 * [944] Delete Columns to Make Sorted
 */

// @lc code=start
/**
 * @param {string[]} strs
 * @return {number}
 */
var minDeletionSize = function (strs) {
    let deleteCount = 0;
    const N = strs.length;
    const M = strs[0].length;
    for (let j = 0; j < M; j++) {
        for (let i = 1; i < N; i++) {
            if (strs[i - 1].charCodeAt(j) > strs[i].charCodeAt(j)) {
                deleteCount++;
                break;
            }
        }
    }

    return deleteCount;
};
// @lc code=end

File: src/2025/944.delete-columns-to-make-sorted.cpp
/*
 * @lc app=leetcode id=944 lang=cpp
 *
 * [944] Delete Columns to Make Sorted
 */

#include <string>
#include <vector>

// @lc code=start
class Solution {
 public:
  int minDeletionSize(const std::vector<std::string>& strs) {
    const std::size_t row_count = strs.size();
    const std::size_t column_count = strs[0].size();
    int deletion_size = 0;
    for (std::size_t column = 0; column < column_count; column++) {
      for (std::size_t row = 1; row < row_count; row++) {
        if (strs[row - 1][column] > strs[row][column]) {
          deletion_size++;
          break;
        }
      }
    }

    return deletion_size;
  }
};
// @lc code=end
